googlemaps/maps.py

```diff
@@ -0,0 +1,252 @@
+)
+
+MAPS_MAP_TYPES = set(
+    ['roadmap', 'satellite', 'terrain', 'hybrid']
+)
+
+
+class StaticMapParam(object):
+    """Base class to handle parameters for Maps Static API."""
+
+    def __init__(self):
+        self.params = []
+
+    def __str__(self):
+        """Converts a list of parameters to the format expected by
+        the Google Maps server.
+
+        :rtype: str
+
+        """
+        return convert.join_list('|', self.params)
+
+
+class StaticMapMarker(StaticMapParam):
+    """Handles marker parameters for Maps Static API."""
+
+    def __init__(self, locations,
+                 size=None, color=None, label=None):
+        """
+        :param locations: Specifies the locations of the markers on
+            the map.
+        :type locations: list
+
+        :param size: Specifies the size of the marker.
+        :type size: str
+
+        :param color: Specifies a color of the marker.
+        :type color: str
+
+        :param label: Specifies a single uppercase alphanumeric
+            character to be displaied on marker.
+        :type label: str
+        """
+
+        super(StaticMapMarker, self).__init__()
+
+        if size:
+            self.params.append("size:%s" % size)
+
+        if color:
+            self.params.append("color:%s" % color)
+
+        if label:
+            if len(label) != 1 or not label.isupper() or not label.isalnum():
+                raise ValueError("Invalid label")
+            self.params.append("label:%s" % label)
+
+        self.params.append(convert.location_list(locations))
+
+
+class StaticMapPath(StaticMapParam):
+    """Handles path parameters for Maps Static API."""
+
+    def __init__(self, points,
+                 weight=None, color=None,
+                 fillcolor=None, geodesic=None):
+        """
+        :param points: Specifies the point through which the path
+            will be built.
+        :type points: list
+
+        :param weight: Specifies the thickness of the path in pixels.
+        :type weight: int
+
+        :param color: Specifies a color of the path.
+        :type color: str
+
+        :param fillcolor: Indicates both that the path marks off a
+            polygonal area and specifies the fill color to use as an
+            overlay within that area.
+        :type fillcolor: str
+
+        :param geodesic: Indicates that the requested path should be
+            interpreted as a geodesic line that follows the curvature
+            of the earth.
+        :type geodesic: bool
+        """
+
+        super(StaticMapPath, self).__init__()
+
+        if weight:
+            self.params.append("weight:%s" % weight)
+
+        if color:
+            self.params.append("color:%s" % color)
+
+        if fillcolor:
+            self.params.append("fillcolor:%s" % fillcolor)
+
+        if geodesic:
+            self.params.append("geodesic:%s" % geodesic)
+
+        self.params.append(convert.location_list(points))
+
+
+def maps_download(client, size,
```

Comment:
what about `def static_map`? this makes it a little easier to link to all other documentation on the api

Reply:
I was thinking a lot about the name of this function and unfortunately the name "static_map" had not came to mind. I agree that it is more suitable name.